chia/full_node/full_node_api.py

```diff
@@ -1460,16 +1460,22 @@ async def register_interest_in_puzzle_hash(
 
         hint_coin_ids = []
         # Add peer to the "Subscribed" dictionary
-        max_items = self.full_node.config.get("max_subscribe_items", 200000)
+        if self.is_trusted(peer):
+            max_items = self.full_node.config.get("trusted_max_subscribe_items", 2000000)
+        else:
+            max_items = self.full_node.config.get("max_subscribe_items", 200000)
         for puzzle_hash in request.puzzle_hashes:
-            ph_hint_coins = await self.full_node.hint_store.get_coin_ids(puzzle_hash)
-            hint_coin_ids.extend(ph_hint_coins)
+
+            hint_coin_ids.extend(await self.full_node.hint_store.get_coin_ids(puzzle_hash))
+
+            # check subscription limit
+            if self.full_node.peer_sub_counter[peer.peer_node_id] >= max_items:
+                self.log.warning(f"wallet {peer.peer_node_id} exceeded max puzzle_hash subscription limit")
```

Comment:
I think this should be `INFO`. It's not a problem for the full node

Reply:
i dont think so, this could be your own wallet, and it really shouldn't happen this means the receiving wallet will miss coins

Reply:
Agree - a user with a large wallet/lots of CATs will most likely be running their own full node. When they hit the limit and things start acting weird, they'll check their logs or support will ask them for the logs.